drivers/can/can_mcux_flexcan.c

```diff
@@ -702,6 +709,13 @@ static int mcux_flexcan_init(const struct device *dev)
 		return err;
 	}
 
+#ifdef CONFIG_PINCTRL
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
```

Comment:
I think `pinctrl_apply_state()` should go before `mcux_flexcan_set_mode()` to ensure the pins are routed to the CAN transceiver before trying to initialize the CAN controller?

Reply:
thanks for pointing that out, I've moved the `pinctrl_apply_state` to before `mcux_flexcan_set_mode`